visualize serverless function in topology

nemesis09 opened the pull request:
Story: 
https://issues.redhat.com/browse/ODC-5480

Problem/Description:
Serverless functions should be visually identifiable on topology

Solution:
use distinct icon and background color to distinctly identify serverless function in topology

Screens:
_Topology_
![Screenshot from 2021-02-24 17-34-04](https://user-images.githubusercontent.com/38663217/108998406-e30fd580-76c6-11eb-87c7-7e9040d92273.png)

_Topology List View_
![Screenshot from 2021-02-25 14-49-15](https://user-images.githubusercontent.com/38663217/109131789-1e6adc80-7779-11eb-868d-11db5a6bfbf3.png)

Test Coverage:
![Screenshot from 2021-02-24 17-36-14](https://user-images.githubusercontent.com/38663217/108998507-020e6780-76c7-11eb-841c-03483872e2e9.png)

Browser Conformation:
- [x] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge

vikram-raj commented on frontend/packages/knative-plugin/src/components/overview/KnativeResourceOverviewPage.tsx:
```diff
@@ -83,12 +84,18 @@ export const KnativeResourceOverviewPage: React.ComponentType<KnativeResourceOve
+  const {
+    metadata: { labels },
+  } = item?.obj;
+  const isServerlessFunction = labels && labels['boson.dev/function'];
```
```suggestion
  const isServerlessFunction = labels?.['boson.dev/function'];
```
and better put label `boson.dev/function` in a const as it has been used multiple places.

nemesis09 replied:
done

vikram-raj commented on frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.scss:
```diff
@@ -0,0 +1,3 @@
+.odc-serverless-function-label {
+    margin-left: var(--pf-global--spacer--md);
+}
```
nit: add a new line here

nemesis09 replied:
done

vikram-raj commented on frontend/packages/knative-plugin/src/components/overview/ServerlessFunctionLabel.tsx:
```diff
@@ -0,0 +1,10 @@
+
+const ServerlessFunctionLabel: React.FC = () => (
+  <Label variant="filled" className="odc-serverless-function-label" color="purple">
+    Function
```
localize the string

nemesis09 replied:
done

vikram-raj commented on frontend/packages/knative-plugin/src/topology/components/groups/KnativeServiceGroup.tsx:
```diff
@@ -118,6 +119,10 @@ const KnativeServiceGroup: React.FC<KnativeServiceGroupProps> = ({
+  const {
+    metadata: { labels },
+  } = getResource(element);
+  const isServerlessFunction = labels && labels['boson.dev/function'];
```
Same here
```suggestion
  const isServerlessFunction = labels?.['boson.dev/function'];
```

nemesis09 replied:
done

nemesis09 commented:
@vikram-raj 
I've updated the PR and addressed the reviews. PTAL

christianvogt commented:
@openshift/team-devconsole-ux @serenamarie125 are we still putting the `function` label in the header?

nemesis09 commented:
/assign @invincibleJai

nemesis09 commented:
/retest

christianvogt commented:
/approve

invincibleJai commented:
Looks good 

cc @vikram-raj

vikram-raj commented:
@nemesis09 tests are failing because `knative-plugin.json` is not updated. Please run `yarn i18n` and update the PR.

nemesis09 commented:
> @nemesis09 tests are failing because `knative-plugin.json` is not updated. Please run `yarn i18n` and update the PR.

updated

vikram-raj commented:
![image](https://user-images.githubusercontent.com/2561818/109051123-34888680-7700-11eb-9df9-afc8a389d612.png)

function icon needs to be updated in the topology list view as well.

nemesis09 commented:
> ![image](https://user-images.githubusercontent.com/2561818/109051123-34888680-7700-11eb-9df9-afc8a389d612.png)
> 
> function icon needs to be updated in the topology list view as well. `nodetest` is knative function.

updated PR addressing the review comment. PTAL

vikram-raj reviewed (commented):
Verified the changes. and it works as expected

/lgtm

vikram-raj commented:
/retest
